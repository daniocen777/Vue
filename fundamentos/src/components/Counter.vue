<template>
  <!-- No tener mucos cálculos -->
  <h2>{{ customTitle }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <div>
    <!-- Atajo => v-on por @ -->
    <button v-on:click="increase">+1</button>
    <button @click="decrease">-1</button>
  </div>
</template>

<script>
/* Es reactivo por defecto */
export default {
  name: "Counter",
  /* Propieadades */
  props: {
    title: String,
    start: {
      type: Number,
      //required: true,
      default: 10,
    },
  },
  /* Estados */
  data() {
    return {
      counter: this.start,
    };
  },
  /* Métodos */
  // Calcular el cuadrado
  methods: {
    getSquareValue() {
      return this.counter * this.counter;
    },
    increase() {
      return this.counter++;
    },
    decrease() {
      return this.counter--;
    },
  },
  /* Proiedades computadas (guaradas en chaché)
  Sólo cambia cuando uno de los datos cambia */
  computed: {
    squareCounter() {
      console.log("squareCounter");
      return this.counter * this.counter;
    },
    customTitle() {
      return this.title || "Contador";
    },
  },
};
</script>

<style>
button {
  background-color: #129087;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 15px;
}

button:hover {
  background-color: #127087;
  transition: 0.3s ease-in-out;
}
</style>