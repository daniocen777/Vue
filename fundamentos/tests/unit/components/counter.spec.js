import { shallowMount } from "@vue/test-utils";

import Counter from "@/components/Counter";

describe("Counter Component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Counter); // Instancia del componente
  });

  /*  */
  test("Establece valor por defecto", () => {
    // Leer props
    const { start } = wrapper.props();

    const value = wrapper.find('[data-testid="counter"]').text();

    expect(Number(value)).toBe(start);
  });

  /* 1° Prueba */
  /* test("Debe ser match con el snapshot (template)", () => {
    const wrapper = shallowMount(Counter); // Instancia del componente
    // Para actualizar el snapshot, ejecutar: npm run test:unit -u
    expect(wrapper.html()).toMatchSnapshot();
  }); */
  /* 2° Prueba => <h2> */
  test("h2 debe de tener valor por defecto 'Contador'", () => {
    const h2 = wrapper.find("h2"); // Encontrr el primer h2
    expect(h2.text()).toBe("Contador");
  });

  /* 3° Prueba*/
  test("El valor por defecto debe ser 15 en <p>", () => {
    const value = wrapper.find('[data-testid="counter"]').text();
    expect(value).toBe("15");
  });
  /* 4° Prueba*/
  test("Debe incrementar y decrementar el valor del contador", async () => {
    const [increaseButton, decreaseButton] = wrapper.findAll("button");
    // Evaluar un evento click (referencia al botón)
    await increaseButton.trigger("click"); // simulando un click
    await increaseButton.trigger("click"); // simulando un click
    await increaseButton.trigger("click"); // simulando un click
    await decreaseButton.trigger("click"); // simulando un click
    await decreaseButton.trigger("click"); // simulando un click

    const value = wrapper.find('[data-testid="counter"]').text();

    expect(value).toBe("16");
  });

  /* Enviar props */
  test("Muestra la prop 'title'", () => {
    const title = "Título de prueba";
    const wraper = shallowMount(Counter, {
      props: {
        title: title,
      },
    });

    expect(wraper.find("h2").text()).toBe(title);
  });
});
