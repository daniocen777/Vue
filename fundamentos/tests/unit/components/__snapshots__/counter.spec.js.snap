// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe ser match con el snapshot (template) 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <!-- No tener mucos cÃ¡lculos -->
      <h2>
        Contador
      </h2>
      <p>
        10 
        <sup>
          2
        </sup>
         = 100
      </p>
      <div>
        <!-- Atajo =&gt; v-on por @ -->
        <button>
          +1
        </button>
        <button>
          -1
        </button>
      </div>
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "counter": 10,
          "customTitle": "Contador",
          "decrease": [Function],
          "getSquareValue": [Function],
          "hasOwnProperty": [Function],
          "increase": [Function],
          "squareCounter": 100,
          "start": 10,
          "title": undefined,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "counter": 2,
            "customTitle": 4,
            "decrease": 4,
            "getSquareValue": 4,
            "hasOwnProperty": 4,
            "increase": 4,
            "isVueInstance": 0,
            "squareCounter": 4,
            "start": 3,
            "title": 3,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "counter": 10,
            "customTitle": "Contador",
            "decrease": [Function],
            "getSquareValue": [Function],
            "hasOwnProperty": [Function],
            "increase": [Function],
            "squareCounter": 100,
            "start": 10,
            "title": undefined,
          },
          "da": null,
          "data": Object {
            "counter": 10,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
                ReactiveEffect {
                  "active": true,
                  "computed": ComputedRefImpl {
                    "__v_isReadonly": false,
                    "__v_isRef": true,
                    "_cacheable": true,
                    "_dirty": false,
                    "_setter": [Function],
                    "_value": 100,
                    "dep": Set {
                      [Circular],
                    },
                    "effect": [Circular],
                  },
                  "deps": Array [
                    [Circular],
                  ],
                  "fn": [Function],
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "start": 10,
            "title": undefined,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "start": Object {
                "0": false,
                "1": true,
                "default": 10,
                "type": [Function],
                "validator": [Function],
              },
              "title": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [
              "start",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "counter": 10,
            "customTitle": "Contador",
            "decrease": [Function],
            "getSquareValue": [Function],
            "hasOwnProperty": [Function],
            "increase": [Function],
            "squareCounter": 100,
            "start": 10,
            "title": undefined,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": 100,
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        Set {
                          [Circular],
                        },
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": "Contador",
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        [Circular],
                        Set {
                          [Circular],
                          ReactiveEffect {
                            "active": true,
                            "computed": ComputedRefImpl {
                              "__v_isReadonly": false,
                              "__v_isRef": true,
                              "_cacheable": true,
                              "_dirty": false,
                              "_setter": [Function],
                              "_value": 100,
                              "dep": Set {
                                [Circular],
                              },
                              "effect": [Circular],
                            },
                            "deps": Array [
                              [Circular],
                            ],
                            "fn": [Function],
                            "parent": undefined,
                            "scheduler": [Function],
                          },
                        },
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                    ReactiveEffect {
                      "active": true,
                      "computed": ComputedRefImpl {
                        "__v_isReadonly": false,
                        "__v_isRef": true,
                        "_cacheable": true,
                        "_dirty": false,
                        "_setter": [Function],
                        "_value": 100,
                        "dep": Set {
                          [Circular],
                        },
                        "effect": [Circular],
                      },
                      "deps": Array [
                        [Circular],
                      ],
                      "fn": [Function],
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": " No tener mucos cÃ¡lculos ",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <!-- No tener mucos cÃ¡lculos -->,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Comment),
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Contador",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h2>
                  Contador
                </h2>,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h2",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "10 ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": 10 ,
                    "key": null,
                    "patchFlag": 1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "2",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <sup>
                      2
                    </sup>,
                    "key": null,
                    "patchFlag": -1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "sup",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " = 100",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  = 100,
                    "key": null,
                    "patchFlag": 1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p>
                  10 
                  <sup>
                    2
                  </sup>
                   = 100
                </p>,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Atajo => v-on por @ ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <!-- Atajo =&gt; v-on por @ -->,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Comment),
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "+1",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button>
                      +1
                    </button>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "-1",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button>
                      -1
                    </button>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div>
                  <!-- Atajo =&gt; v-on por @ -->
                  <button>
                    +1
                  </button>
                  <button>
                    -1
                  </button>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Contador",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h2>
                  Contador
                </h2>,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h2",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "10 ",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": 10 ,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Text),
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": " = 100",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el":  = 100,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 8,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Text),
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(Fragment),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "computed": Object {
              "customTitle": [Function],
              "squareCounter": [Function],
            },
            "data": [Function],
            "methods": Object {
              "decrease": [Function],
              "getSquareValue": [Function],
              "increase": [Function],
            },
            "name": "Counter",
            "props": Object {
              "start": Object {
                "0": false,
                "1": true,
                "default": 10,
                "type": [Function],
                "validator": [Function],
              },
              "title": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "computed": Object {
            "customTitle": [Function],
            "squareCounter": [Function],
          },
          "data": [Function],
          "methods": Object {
            "decrease": [Function],
            "getSquareValue": [Function],
            "increase": [Function],
          },
          "name": "Counter",
          "props": Object {
            "start": Object {
              "0": false,
              "1": true,
              "default": 10,
              "type": [Function],
              "validator": [Function],
            },
            "title": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.31",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "counter": 10,
    "customTitle": "Contador",
    "decrease": [Function],
    "getSquareValue": [Function],
    "hasOwnProperty": [Function],
    "increase": [Function],
    "squareCounter": 100,
    "start": 10,
    "title": undefined,
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": ,
}
`;
